src/components/Api.js

```diff
@@ -0,0 +1,123 @@
+class Api {
+    constructor({baseUrl, authToken, headers}) {
+        this._baseUrl = baseUrl;
+        this._authToken = authToken;
+        this._headers = headers;
+    }
+
+    getInitialCards() {
+        return fetch("https://around-api.en.tripleten-services.com/v1/cards",)
+    }
+
+        //methods for working with the API 
+        
+        //GET https://around.nomoreparties.co/v1/groupID/cards
+             getCardList(){
+                return fetch(`${this._baseUrl}/cards`, {
+                    headers: {
+                      authorization: this._authToken
+                    }
+                  })
+                    .then(res => {res.ok ? res.json() : Promise.reject(`Error: ${res.status}`)})
+                    .catch((err) => {
```

Comment:
Needs correction: Errors should be handled outside of the api class in `index.js`. Now `catch` block is placed in the middle of the chain, so the rest of the chain won't know anything about the error — but it should because we need to update the data only in case of success

The api call in `index.js` should look like this:

```js
// show loading
api.
  .then(...)
  .catch(...)
  .finally(() => {
    // hide loading
  })
```